tests/ctags/var-and-return-type.cpp

```diff
@@ -0,0 +1,40 @@
+
+const volatile unsigned int func1();
+const volatile unsigned int var1 = 0;
+
+struct type1 {
+	unsigned int memb1;
+	struct type1 *next;
+};
+
+const struct type1 func2();
+const struct type1 var2 = { 0, 0 };
+
+typedef type1 type1_t;
+
+const type1_t func3();
+const type1_t var3 = { 0, 0 };
+
+struct type1 func4();
+struct type1 var4 = { 0, 0 };
+
+type1_t func5();
+type1_t var5 = { 0, 0 };
+
```

Comment:
> Nothing to do with reporting the var type, it's "just" another mean of initializing. So different issue, possibly different test case.

Fair enough

> AFAIK -> syntax is not supported at all yet.

For some reason I thought it was, maybe another language.

> (OT, WTF is about this syntax? why add a new one?)

So it is after parameter declaration and so can use decltype(parameter) where the parameter type is from a template parameter and so not known until template parameter deduction.